projects/common-platform-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/messaging/Handler.kt

```
@@ -1,29 +1,82 @@
+import uk.gov.justice.digital.hmpps.telemetry.TelemetryService
 
 @Component
 @Channel("common-platform-and-delius-queue")
 class Handler(
-    override val converter: NotificationConverter<HmppsDomainEvent>,
-    private val telemetryService: TelemetryService
-) : NotificationHandler<HmppsDomainEvent> {
-    @Publish(
-        messages = [
-            // TODO list the event types here that this service will subscribe to. For example,
-            // Message(name = "approved-premises/application-assessed"),
-            // Message(title = "probation-case.prison-identifier.added"),
-        ]
-    )
-    override fun handle(notification: Notification<HmppsDomainEvent>) {
+    override val converter: NotificationConverter<CommonPlatformHearing>,
+    private val notifier: Notifier,
+    private val telemetryService: TelemetryService,
+    private val personService: PersonService,
+    private val probationSearchClient: ProbationSearchClient
+) : NotificationHandler<CommonPlatformHearing> {
+
+    @Publish(messages = [Message(title = "COMMON_PLATFORM_HEARING", payload = Schema(CommonPlatformHearing::class))])
+    override fun handle(notification: Notification<CommonPlatformHearing>) {
         telemetryService.notificationReceived(notification)
-        TODO("Not yet implemented")
+
+        // Filter hearing message for defendants with a convicted judicial result of Remanded in custody
+        val defendants = notification.message.hearing.prosecutionCases
+            .flatMap { it.defendants }
+            .filter { defendant ->
+                defendant.offences.any { offence ->
+                    offence.judicialResults.any { judicialResult ->
+                        judicialResult.isConvictedResult == true && judicialResult.label == "Remanded in custody"
+                    }
+                }
+            }
+            .ifEmpty { throw IllegalArgumentException("No valid defendants found") }
```

Do we need to throw an exception here?